fdbclient/include/fdbclient/FDBTypes.h

```diff
@@ -1595,24 +1610,38 @@ struct StorageMetadataType {
 	// when the SS is initialized, in epoch seconds, comes from currentTime()
 	double createdTime;
 	KeyValueStoreType storeType;
+	bool needReplacement = false;
+	std::set<std::string> paramsNeedTobeReplaced;
 
 	// no need to serialize part (should be assigned after initialization)
 	bool wrongConfigured = false;
 
 	StorageMetadataType() : createdTime(0) {}
-	StorageMetadataType(double t, KeyValueStoreType storeType = KeyValueStoreType::END, bool wrongConfigured = false)
-	  : createdTime(t), storeType(storeType), wrongConfigured(wrongConfigured) {}
+	StorageMetadataType(double t,
+	                    KeyValueStoreType storeType = KeyValueStoreType::END,
+	                    bool wrongConfigured = false,
+	                    bool needReplacement = false,
+	                    const std::set<std::string>& params = {})
+	  : createdTime(t), storeType(storeType), wrongConfigured(wrongConfigured), needReplacement(needReplacement),
+	    paramsNeedTobeReplaced(params) {}
 
 	static double currentTime() { return g_network->timer(); }
 
 	bool operator==(const StorageMetadataType& b) const {
-		return createdTime == b.createdTime && storeType == b.storeType && wrongConfigured == b.wrongConfigured;
+		return createdTime == b.createdTime && storeType == b.storeType && wrongConfigured == b.wrongConfigured &&
+		       needReplacement == b.needReplacement && paramsNeedTobeReplaced.size() == b.paramsNeedTobeReplaced.size();
```

Comment:
Not sure whether just compare params size is enough here. It depends on how the == operation will be use. We'd best put some NOTE comment here?

Reply:
Done.